ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -777,64 +723,56 @@ S3AddReference(WT_STORAGE_SOURCE *storageSource)
     return (0);
 }
 
-/*
- * S3Terminate --
- *     Discard any resources on termination.
- */
+// S3Terminate --
+//    Discard any resources on termination.
+
 static int
 S3Terminate(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session)
 {
-    S3_STORAGE *s3 = (S3_STORAGE *)storageSource;
+    S3Storage *s3 = (S3Storage *)storageSource;
 
     if (--s3->referenceCount != 0)
         return (0);
 
-    /*
-     * Is it currently unclear at the moment what the multi-threading will look like in the
-     * extension. The current implementation is NOT thread-safe, and needs to be addressed in the
-     * future, as mulitple threads could call terminate leading to a race condition.
-     */
+    // It is currently unclear at the moment what the multi-threading will look like in the
+    // extension. The current implementation is NOT thread-safe, and needs to be addressed in the
+    // future, as multiple threads could call terminate leading to a race condition.
     while (!s3->fhList.empty()) {
-        S3_FILE_HANDLE *fs = s3->fhList.front();
+        S3FileHandle *fs = s3->fhList.front();
         S3FileClose((WT_FILE_HANDLE *)fs, session);
     }
-    /*
-     * Terminate any active filesystems. There are no references to the storage source, so it is
-     * safe to walk the active filesystem list without a lock. The removal from the list happens
-     * under a lock. Also, removal happens from the front and addition at the end, so we are safe.
-     */
+
+    // Terminate any active filesystems. There are no references to the storage source, so it is
+    // safe to walk the active filesystem list without a lock. The removal from the list happens
+    // under a lock. Also, removal happens from the front and addition at the end, so we are safe.
     while (!s3->fsList.empty()) {
-        S3_FILE_SYSTEM *fs = s3->fsList.front();
+        S3FileSystem *fs = s3->fsList.front();
         S3FileSystemTerminate(&fs->fileSystem, session);
     }
 
-    /* Log collected statistics on termination. */
-    S3ShowStatistics(*s3);
+    S3LogStatistics(*s3);
 
     Aws::Utils::Logging::ShutdownAWSLogging();
     Aws::ShutdownAPI(options);
 
+    s3->log->LogDebugMessage("S3Terminate: Terminating S3 storage source.");
```

Comment:
Terminating --> Terminated. OR move the message to the start of the function.

Reply:
Fixed.